arangod/GeneralServer/H2CommTask.cpp

```diff
@@ -366,40 +372,36 @@ void H2CommTask<T>::upgradeHttp1(std::unique_ptr<HttpRequest> req) {
   }
 
   // https://http2.github.io/http2-spec/#discover-http
-  auto str = std::make_shared<std::string>();
-  str->append(
-      "HTTP/1.1 101 Switching Protocols\r\nConnection: "
-      "Upgrade\r\nUpgrade: h2c\r\n\r\n");
-
-  auto buffer = asio_ns::buffer(str->data(), str->size());
+  auto buffer =
+      asio_ns::buffer(::switchingProtocols.data(), ::switchingProtocols.size());
   asio_ns::async_write(
       this->_protocol->socket, buffer,
-      withLogContext([self(this->shared_from_this()), str(std::move(str)),
-                      req(std::move(req))](const asio_ns::error_code& ec,
-                                           std::size_t) mutable {
-        auto& me = static_cast<H2CommTask<T>&>(*self);
-        if (ec) {
-          me.close(ec);
-          return;
-        }
+      withLogContext(
+          [self(this->shared_from_this()), req(std::move(req))](
+              asio_ns::error_code const& ec, std::size_t nwrite) mutable {
+            auto& me = static_cast<H2CommTask<T>&>(*self);
+            if (ec) {
+              me.close(ec);
+              return;
+            }
 
-        submitConnectionPreface(me._session);
+            submitConnectionPreface(me._session);
 
-        // The HTTP/1.1 request that is sent prior to upgrade is assigned
-        // a stream identifier of 1 (see Section 5.1.1).
-        // Stream 1 is implicitly "half-closed" from the client toward the
-        // server
+            // The HTTP/1.1 request that is sent prior to upgrade is assigned
+            // a stream identifier of 1 (see Section 5.1.1).
+            // Stream 1 is implicitly "half-closed" from the client toward the
+            // server
 
-        TRI_ASSERT(req->messageId() == 1);
-        auto* strm = me.createStream(1, std::move(req));
-        TRI_ASSERT(strm);
+            TRI_ASSERT(req->messageId() == 1);
+            auto* strm = me.createStream(1, std::move(req));
```

Comment:
Btw is every http 1.1 connection will be blocked on single mutex here?
Is it really ok for us?